### ISSUE TITLE
Slow Rename (Move) Process (Unraid)(Docker) 

### ISSUE THREAD

**Issue by restjack** [jlesage/docker-filebot#15]
When Renaming any File it can take upto a minute per file to rename it.
all files are local its going from download folder to Video Folder 
Download Path /mnt/user/Downloads/Jdownloader/
Video Path /mnt/user/Video/
![Screenshot (213)](https://user-images.githubusercontent.com/5397315/71315720-d768b500-2418-11ea-975a-6bdd634fa5d7.png)

Renaming Variables
/Video/Tv shows/{ny}/{n} - {s00e00} - {t} [{resolution}] {video.CodecID =~ /HEVC/ ? 'HEVC' : 'Not HEVC'} {bitdepth}Bit {ac} {channels}

When using filebot on windows the rename process is nearly instant.

**Comment by jlesage** [comment]
Note that the file is not simply renamed, but it is moved to different folder.
And because your input and output folders are on different shares, moving the files takes longer (the file is in fact copied then deleted from the source).

**Comment by restjack** [comment]
Both Directories are on the Same Disk (Share)
Disk HDD 6  /mnt/user/Downloads/Jdownloader/
Moved to 
Disk HDD 6 /mnt/user/Video/
just different parent folders (from my understanding)
is there any changes I can Make to Make this work?
without it having to copy the file then deleting the original ?

**Comment by jlesage** [comment]
Even if both folders are on the same physical disc, they are on different shares, which is handled differently by the underlying filesystem.

Maybe I'm wrong, but to confirm you could try to rename on the same share (e.g. from `/mnt/user/Downloads/Jdownloader/` to `/mnt/user/Downloads/Video/`) to see if you see any improvement.

**Comment by restjack** [comment]
Thank you for the Feedback I have had success in getting it working by 
changing the Downloads & Video Folder in the docker Template to both point to the 
/mnt/cache/

and inside the docker changed my script to /Video/Video/Tv shows/{ny}/{n} - {s00e00} - {t} [{resolution}] {video.CodecID =~ /HEVC/ ? 'HEVC' : 'Not HEVC'} {bitdepth}Bit {ac} {channels}

Using /video/ 
as the source of the files 
it now renames near instant.
thank you 
![Screenshot (214)](https://user-images.githubusercontent.com/5397315/71398747-4f300e80-25d6-11ea-9fd1-479c1916e5c7.png)

**Comment by WirlyWirly** [comment]
I'm also seeing much slower copy rates than I'm used. Thanks for the explanation. Makes sense, although 5x slower still seems excessive. Guess that's the trade-off of dockers!

**Comment by jlesage** [comment]
Docker is probably not the reason for the slow copy.
Are you using Docker under Windows ?

**Comment by WirlyWirly** [comment]
Yes I am. I did some speed comparisons and these were my results copying a 1gb video file...

Using windows explorer (drag and drop): 6.5s
sh into the container and run `cp /media/input/video.mkv /media/output/`: 18s
AMC: 1m48s

So it's actually 16x slower than explorer but only 18 slower than using cp after I sh into the container.

**Comment by jlesage** [comment]
On Windows, Docker runs in a Linux VM... so this explain a part of the issue.
Also, the way Filebot copies the files could also have an effect.  If you try the Windows version of Filebot, you may also see that copies are slower than Explorer.

**Comment by SlashGIP** [comment]
Hi, i got a similar thing here. 

I have a Linux VPS with mounted Storage from a provider. Say google or something. 

Now I mounted the drive "googledrive". In that "googledrive" are two directorys.. "one" and "two". Now I stored the unnamed files in "one" and want to move (rename) it automaticly with filebot (I have a licence) into the dir "two". Now filebot first download the file from "one", rename it, and upload it into "two" with the new name. Why this is happening? 

On my LAN I mount the same drives in the same way to give it free with samba. My Windows Laptop, where i gave these two directorys X: and Y: with the GUI have no problem to take the files from the directory "one" and rename them directly into  directory "two". 

I do not understand it. Maybe somebody can show me wheres my fault here... 

Thx a lot
